dali_tf_plugin/daliop.cc

```diff
@@ -159,6 +162,28 @@ class DaliOp : public tf::OpKernel {
   }
 
   ~DaliOp() override {
+    if (get_memory_stats_) {
+      size_t N;
+      daliExecutorMetadata *meta;
+      daliGetExecutorMetadata(&pipe_handle_, &meta, &N);
+      std::cout << "DALI operator memory statistics: "  << std::endl;
+      for (size_t i = 0; i < N; ++i) {
+        std::cout << "Operator " << meta[i].operator_name;
+        for (size_t j = 0; j < meta[i].out_num; ++j) {
+          std::cout << "   output [ " << j << " ] : "
+                    << meta[i].real_size[j] << "B allocated "
+                    << meta[i].reserved[j] << "B reserved";
+          if (j != meta[i].out_num - 1) {
+            std::cout << ",";
+          }
+        }
+        std::cout << std::endl;
+        free(meta[i].operator_name);
```

Comment:
here we would call the deleter function I suggested

Reply:
Done